# Housing Market

### Introduction:

This time we will create our own dataset with fictional numbers to describe a house market. As we are going to create random data don't try to reason of the numbers.

### Step 1. Import the necessary libraries

In [1]:
import pandas as pd
import numpy as np

### Step 2. Create 3 differents Series, each of length 100, as follows: 
1. The first a random number from 1 to 4 
2. The second a random number from 1 to 3
3. The third a random number from 10,000 to 30,000

In [3]:
s1 = pd.Series(np.random.randint(1, 5, size=100))

s2 = pd.Series(np.random.randint(1, 4, size=100))

s3 = pd.Series(np.random.randint(10000, 30001, size=100))

In [4]:
print(s1.head())
print(s2.head())
print(s3.head())

0    2
1    2
2    1
3    1
4    2
dtype: int32
0    1
1    1
2    2
3    3
4    3
dtype: int32
0    16718
1    14476
2    20442
3    10016
4    28703
dtype: int32


### Step 3. Let's create a DataFrame by joinning the Series by column

In [5]:
df = pd.concat([s1, s2, s3], axis=1)

In [ ]:
df.columns = ['0', '1', '2'] 

In [7]:
print(df.head())

   0  1      2
0  2  1  16718
1  2  1  14476
2  1  2  20442
3  1  3  10016
4  2  3  28703


### Step 4. Change the name of the columns to bedrs, bathrs, price_sqr_meter

In [8]:
df.columns = ['bedsr', 'baths', 'price_sqr_meter'] #Renombramos

In [9]:
print(df.head())

   bedsr  baths  price_sqr_meter
0      2      1            16718
1      2      1            14476
2      1      2            20442
3      1      3            10016
4      2      3            28703


### Step 5. Create a column DataFrame with the values of the 3 Series and assign it to 'bigcolumn'

In [10]:
bigcolumn = pd.DataFrame(pd.concat([s1, s2, s3], ignore_index=True), columns=['Values'])

In [13]:
print(bigcolumn)

     Values
0         2
1         2
2         1
3         1
4         2
..      ...
295   22955
296   19150
297   21614
298   12260
299   15118

[300 rows x 1 columns]


In [15]:
bigcolumn = pd.DataFrame(pd.concat([s1, s2, s3], ignore_index=True), columns=[0])
print(bigcolumn)

         0
0        2
1        2
2        1
3        1
4        2
..     ...
295  22955
296  19150
297  21614
298  12260
299  15118

[300 rows x 1 columns]


### Step 6. Oops, it seems it is going only until index 99. Is it true?

Sí, surge un array en donde  los índices se reinician después de 99, porque al concatenar las tres Series de longitud 100 cada una, se usó ignore_index=True , y al concatenarlas se formó un array continuo de 300 elementos, pero los índices visibles que se muestran corresponden a la posición relativa dentro de cada Serie, por eso se repiten de 0 a 99 tres veces, aunque existan 300.

In [16]:
bigcolumn = pd.DataFrame(pd.concat([s1, s2, s3], ignore_index=True))
print(bigcolumn.index)

RangeIndex(start=0, stop=300, step=1)


In [17]:
bigcolumn = pd.concat([s1, s2, s3], ignore_index=True)
bigcolumn.index.to_numpy()

array([  0,   1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,
        13,  14,  15,  16,  17,  18,  19,  20,  21,  22,  23,  24,  25,
        26,  27,  28,  29,  30,  31,  32,  33,  34,  35,  36,  37,  38,
        39,  40,  41,  42,  43,  44,  45,  46,  47,  48,  49,  50,  51,
        52,  53,  54,  55,  56,  57,  58,  59,  60,  61,  62,  63,  64,
        65,  66,  67,  68,  69,  70,  71,  72,  73,  74,  75,  76,  77,
        78,  79,  80,  81,  82,  83,  84,  85,  86,  87,  88,  89,  90,
        91,  92,  93,  94,  95,  96,  97,  98,  99, 100, 101, 102, 103,
       104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116,
       117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129,
       130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142,
       143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155,
       156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168,
       169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 18

In [18]:
array_s1 = s1.to_numpy()
print(array_s1)

[2 2 1 1 2 2 2 4 3 2 1 1 1 2 3 4 2 2 3 4 1 4 1 2 4 2 2 2 1 3 1 2 1 3 2 1 1
 4 2 1 3 2 1 4 4 4 4 2 2 1 3 4 3 4 2 3 2 2 2 4 4 4 1 3 3 3 1 1 4 3 1 3 4 4
 1 4 2 4 4 1 3 1 3 1 1 1 3 4 3 1 2 1 4 1 3 4 2 2 2 3]


In [19]:
#Lo que s ehizo en el Output fue:
s1.index = np.arange(1, 101)

print(s1)

1      2
2      2
3      1
4      1
5      2
      ..
96     4
97     2
98     2
99     2
100    3
Length: 100, dtype: int32


In [20]:
indices_array = np.arange(len(s1))  # Del 0 al 99
print(indices_array)

[ 0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23
 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47
 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71
 72 73 74 75 76 77 78 79 80 81 82 83 84 85 86 87 88 89 90 91 92 93 94 95
 96 97 98 99]


### Step 7. Reindex the DataFrame so it goes from 0 to 299

In [23]:
bigcolumn = bigcolumn.reindex(range(300))
bigcolumn

0          2
1          2
2          1
3          1
4          2
       ...  
295    22955
296    19150
297    21614
298    12260
299    15118
Length: 300, dtype: int32

In [26]:
bigcolumn = bigcolumn.reindex(range(300), fill_value=0)
bigcolumn

0          2
1          2
2          1
3          1
4          2
       ...  
295    22955
296    19150
297    21614
298    12260
299    15118
Length: 300, dtype: int32

In [21]:
bigcolumn = bigcolumn.reindex(range(300))
bigcolumn

0          2
1          2
2          1
3          1
4          2
       ...  
295    22955
296    19150
297    21614
298    12260
299    15118
Length: 300, dtype: int32

In [ ]:
bigcolumn = pd.concat([s1, s2, s3], ignore_index=True)
bigcolumn.index.to_numpy()

array([  0,   1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,
        13,  14,  15,  16,  17,  18,  19,  20,  21,  22,  23,  24,  25,
        26,  27,  28,  29,  30,  31,  32,  33,  34,  35,  36,  37,  38,
        39,  40,  41,  42,  43,  44,  45,  46,  47,  48,  49,  50,  51,
        52,  53,  54,  55,  56,  57,  58,  59,  60,  61,  62,  63,  64,
        65,  66,  67,  68,  69,  70,  71,  72,  73,  74,  75,  76,  77,
        78,  79,  80,  81,  82,  83,  84,  85,  86,  87,  88,  89,  90,
        91,  92,  93,  94,  95,  96,  97,  98,  99, 100, 101, 102, 103,
       104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116,
       117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129,
       130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142,
       143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155,
       156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168,
       169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 18

In [27]:
print(bigcolumn.dtypes)

int32


In [22]:
bigcolumn.describe()

count      300.000000
mean      6792.700000
std      10122.215642
min          1.000000
25%          2.000000
50%          3.000000
75%      16456.750000
max      29663.000000
dtype: float64